### DISCUSSION TITLE
Support Terms and Conditions

### DISCUSSION THREAD

**Discussion by dcyoung** [clerk/javascript#2570] [Feature Requests]
Our team has implemented flows in multiple apps for gating access until users have signed the latest Terms & Conditions, as well as storing receipts of each confirmation. If this feature was offered by Clerk it would remove much of the need for custom middleware routing and would keep user related metadata in clerk.

Basic parts of such a product feature might be 

- dashboard + API support for updating the version of a terms & conditions for the Clerk App 
- support for checking the latest terms & conditions signed by users 
- clerk middleware support for gating access based on terms 
- control of clerk login redirect behavior if user has not signed latest terms
- potentially offer simple OOTB frontend for accepting terms

**Comment by jescalan** [comment]
You can upvote and drop feedback on this feature [right here](https://feedback.clerk.com/roadmap/c02a065f-e30c-481b-b152-9a0ff41c6f2f)!

**Comment by jescalan** [comment]
Just wanted to provide a short update that this feature was [released in November 2024](https://clerk.com/changelog/2024-11-11-legal-consent), and [documentation is available for it here](https://clerk.com/docs/authentication/configuration/legal-compliance)!
